src/libs/ajax/data-table-providers/WDSDataTableProvider.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import { DataTableFeatures, DataTableProvider, DeleteTableFn, DownloadTsvFn, EntityMetadata, EntityQueryOptions, EntityQueryResponse, GetMetadataFn, GetPageFn } from 'src/libs/ajax/data-table-providers/DataTableProvider'
+
+// interface definitions for WDS payload responses
+interface AttributeSchema {
+  name: string,
+  datatype: string,
+  relatesTo?: string
+}
+
+export interface RecordTypeSchema {
+  name: string,
+  count: number,
+  attributes: AttributeSchema[]
+}
+
+interface SearchRequest {
+  offset: number,
+  limit: number,
+  sort: 'asc' | 'ASC' | 'desc' | 'DESC',
+  sortAttribute: string
+}
+
+interface RecordResponse {
+  id: string,
+  type: string,
+  attributes: { [index: string]: any } // truly "any" here; the backend Java representation is Map<String, Object>
+}
+
+export interface RecordQueryResponse {
+  searchRequest: SearchRequest,
+  totalRecords: number,
+  records: RecordResponse[]
+}
+
+
+export class WDSDataTableProvider implements DataTableProvider {
```

Comment:
I highly recommend that a WdsDataTableProvider interface be defined that extends DataTableProvider and lists the additional public method signatures, etc.  like transformPage method below.

This greatly assists with mocking classes like this when they are the dependency of something else being unit tested.

Also, nit: on the capitalization of WDSDataTableProvider here.... Achrnonyms longer then 2 letters should go cammel-case:  WdsDataTableProvider.